app.py

```diff
@@ -0,0 +1,97 @@
+import panel as pn
+import hvplot.pandas
+
+# Load Data
+from scripts.data import per_dataset_df, all_framework_df
+from scripts.widgets import create_selectwidget, create_togglewidget, create_numberwidget
+from scripts.plots import create_hvplot,create_table
+
+#clean up framework names
+original_framework_names = all_framework_df['framework']
+framework_names = per_dataset_df['framework'].str.extract(r"^(.*?)(?:_|$)")[0]
+per_dataset_df.framework = framework_names
+all_framework_df.framework = framework_names
+
+dataset_list = sorted(list(set(per_dataset_df['dataset'].to_list())))
+
+# Make DataFrame Pipeline Interactive
+per_dataset_idf = per_dataset_df.interactive()
+all_framework_idf = all_framework_df.interactive()
+
+# Define Panel widgets
+metrics = ['loss_rescaled', 'time_train_s_rescaled','time_infer_s_rescaled', 'bestdiff']
+graph_types = ['line', 'bar','hist', 'scatter']
+frameworks_widget = create_selectwidget('Framework', options=framework_names.to_list())
+yaxis_widget =  create_selectwidget('Y-axis Metric', options=metrics)
+yaxis_widget2 =  create_selectwidget('Y-axis Metric', options=metrics)
+datasets_widget = create_togglewidget('Datasets', dataset_list)
+dataset_dropdown = create_selectwidget('Datasets', options=dataset_list)
+graph_type = create_selectwidget('Graph Type', 'bar', graph_types)
+graph_type2 = create_selectwidget('Graph Type', 'bar', graph_types)
+
+#Some data processing
```

Comment:
Create function and add docstring on what the function is trying to do. In this case, what this data processing is about